static/app/views/issueDetails/streamline/issueCronCheckTimeline.tsx

```diff
@@ -0,0 +1,243 @@
+import {Fragment, useMemo, useRef} from 'react';
+import styled from '@emotion/styled';
+
+import {CheckInPlaceholder} from 'sentry/components/checkInTimeline/checkInPlaceholder';
+import {CheckInTimeline} from 'sentry/components/checkInTimeline/checkInTimeline';
+import {
+  Gridline,
+  GridLineLabels,
+  GridLineOverlay,
+} from 'sentry/components/checkInTimeline/gridLines';
+import {useTimeWindowConfig} from 'sentry/components/checkInTimeline/hooks/useTimeWindowConfig';
+import type {StatsBucket} from 'sentry/components/checkInTimeline/types';
+import {Flex} from 'sentry/components/container/flex';
+import {Tooltip} from 'sentry/components/tooltip';
+import {tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Event} from 'sentry/types/event';
+import type {Group} from 'sentry/types/group';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import {useDebouncedValue} from 'sentry/utils/useDebouncedValue';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useUser} from 'sentry/utils/useUser';
+import {useIssueDetails} from 'sentry/views/issueDetails/streamline/context';
+import {getGroupEventQueryKey} from 'sentry/views/issueDetails/utils';
+import {MonitorIndicator} from 'sentry/views/monitors/components/monitorIndicator';
+import {CheckInStatus, type MonitorBucket} from 'sentry/views/monitors/types';
+import {
+  checkInStatusPrecedent,
+  statusToText,
+  tickStyle,
+} from 'sentry/views/monitors/utils';
+import {selectCheckInData} from 'sentry/views/monitors/utils/selectCheckInData';
+import {useMonitorStats} from 'sentry/views/monitors/utils/useMonitorStats';
+
+export function useCronIssueAlertId({groupId}: {groupId: string}): string | undefined {
+  /**
+   * This should be removed once the cron rule value is set on the issue.
+   * This will fetch an event from the max range if the detector details
+   * are not available (e.g. time range has changed and page refreshed)
+   */
+  const user = useUser();
+  const organization = useOrganization();
+  const {detectorDetails} = useIssueDetails();
+  const {detectorId, detectorType} = detectorDetails;
+
+  const hasCronDetector = detectorId && detectorType === 'cron_monitor';
+
+  const {data: event} = useApiQuery<Event>(
+    getGroupEventQueryKey({
+      orgSlug: organization.slug,
+      groupId,
+      eventId: user.options.defaultIssueEvent,
+      environments: [],
+    }),
```

Comment:
does this make another request if environment is set? can you filter these by environment? I think we switched a few places to use `useGroupEvent` to keep the query parameters in sync

Reply:
@scttcper So this is intentionally avoiding the query parameters because the event is purely being used here to have the best shot of getting a valid `monitor.id` tag value. Even if you change the environment, we don't use the new event's `monitor.id` value since they are meant to be identical across events.

This is honestly a work around since this sort of data should just be in the issue payload (but it isn't), but doing it this way does prevent the weird UX of selecting an environment or making a query which unloads the graph (since it'll do it's best to find the monitor ID and put some data on screen.